include/envoy/event/schedulable_cb.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+namespace Event {
+
+/**
+ * Callback wrapper that allows direct scheduling of callbacks in the event loop.
+ */
+class SchedulableCallback {
+public:
+  virtual ~SchedulableCallback() = default;
+
+  /**
+   * Schedule the callback so it runs in the current iteration of the event loop.
+   */
```

Comment:
Are there any ordering guarantees that this API provides? If so it would be good to document them.

Reply:
Done.

Note that the semantics for scheduleCallbackCurrentIteration and scheduleCallbackNextIteration (which I will be introducing in the next PR) are very different.  scheduleCallbackNextIteration will provide little to no ordering guarantees since the target implementation based on timers at 0ms provides no ordering guarantees (items go through an unordered priority queue)